go/vt/sqlparser/expression_rewriting_test.go

```diff
@@ -55,14 +55,19 @@ func TestRewrites(in *testing.T) {
 			db:       true, liid: true,
 		},
 		{
-			in:       "select (select database() from test) from test",
-			expected: "select (select :__vtdbname as `database()` from test) as `(select database() from test)` from test",
+			in:       "select (select database() from dual) from dual",
+			expected: "select (select :__vtdbname as `database()` from dual) as `(select database() from dual)` from dual",
```

Comment:
what would happen with a query like:
`select (select database() from dual) from test`
or
`select (select database()) from test`
or even
`select database() from test`

On a side note, the aliases we are producing look weird to me when the SelectExpr is a function or a query.